ksqldb-common/src/main/java/io/confluent/ksql/util/ConsistencyOffsetVector.java

```diff
@@ -214,8 +223,14 @@ public boolean equals(final Object o) {
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    final ConsistencyOffsetVector that = (ConsistencyOffsetVector) o;
-    return this.dominates(that) && that.dominates(this);
+    final ConsistencyOffsetVector other = (ConsistencyOffsetVector) o;
+
+    try {
+      rwLock.readLock().lock();
```

Comment:
Currently, `equals` is only used in unit tests but i'd prefer if it is correct in production code nevertheless

Reply:
Yeah, this was kind of the point, though. It's not actually correct for production code. We don't grab the other's lock. And the hashCode implementation is also incorrect because the fields it uses are mutable.

Anyway, I think it's not a big deal for this PR, since it is "correct enough" for the current (testing only) use cases. We should probably refactor this class's mutability/locking mechanism in the near future anyway.